### PR TITLE
Code Quality: Added a logger to send exception reports to Sentry

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Opened Files ...
2. ...

### CODE REVIEW

File: src/Files.App/Helpers/Application/AppLifecycleHelper.cs [hunk 1]
```diff
@@ -265,14 +267,6 @@ public static void HandleAppUnhandledException(Exception? ex, bool showToastNoti
...
 				ex.Data[Mechanism.MechanismKey] = "Application.UnhandledException";
 
-				SentrySdk.CaptureException(ex, scope =>
-				{
-					scope.User.Id = generalSettingsService.UserId;
-				});
-
-
-				SentrySdk.CaptureException(ex);
-
```
Comment (hishitetsu) [inline src/Files.App/Helpers/Application/AppLifecycleHelper.cs:275]: @yaira2 is there any reason to capture the exception twice?

**Review by 0x5bfa — COMMENTED** [review]
Sentry logger can be moved to Shared project?

Also, I wanna remove Utilities namespace and replace it with Services(current logger implementation requires major refactoring so not now imo).

**Comment by hishitetsu** [comment]
> Sentry logger can be moved to Shared project?

Should Shared project have a dependency on Sentry?

File: src/Files.App/Helpers/Application/AppLifecycleHelper.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Helpers/Application/AppLifecycleHelper.cs:275]: No, this was an oversight.

**Review by yaira2 — APPROVED** [review]

**Comment by 0x5bfa** [comment]
In order to log in other projects as well.
There's request to log in BackgroundTasks project.

**Comment by yaira2** [comment]
@0x5bfa in theory, we can put `SentryLogger.cs` and the dependency in a shared project.

File: src/Files.App/Utils/Logger/SentryLogger.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			{
+				LogLevel.Debug => SentryLevel.Debug,
+				LogLevel.Information => SentryLevel.Info,
+				LogLevel.Warning => SentryLevel.Warning,
+				LogLevel.Error => SentryLevel.Error,
+				LogLevel.Critical => SentryLevel.Fatal,
+				_ => SentryLevel.Debug
+			};
+
+			SentrySdk.CaptureException(exception, scope =>
```
Comment (yaira2) [inline src/Files.App/Utils/Logger/SentryLogger.cs:39]: Is this data being included for crashes?

```csharp
ex.Data[Mechanism.HandledKey] = false;
ex.Data[Mechanism.MechanismKey] = "Application.UnhandledException";
```

Comment (hishitetsu) [reply src/Files.App/Utils/Logger/SentryLogger.cs:39]: Yes. I left that code.

**Pull request merged**